ci/install-protobuf.sh

```diff
@@ -1,12 +1,35 @@
-#!/bin/sh
+#!/bin/bash
+# Detect OS and architecture
+arch=$(uname -m)
+if [[ $(uname -s) == "Linux" ]]; then
+    os_arch="linux-${arch}";
+elif [[ $(uname -s) == "Darwin" ]]; then # MacOS
+    os_arch="osx-${arch}";
+elif [[ "$OSTYPE" == "cygwin" || "$OSTYPE" == "msys" || "$OSTYPE" == "win32" ]]; then
+    os_arch="win32";
+else
+    echo "unsupported OS, protoc not installed"
+    exit 1;
+fi
+protoc_url=https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protoc-3.6.1-${os_arch}.zip
+
 # Make sure you grab the latest version
-curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip
+echo "fetching ${protoc_url}"
+curl -L -o "protoc.zip" ${protoc_url}
```

Comment:
what if this fails? maybe put in && or $? && cmd

Reply:
solved in 597e6a8